admin-dev/themes/new-theme/js/pages/product/components/categories/tags.ts

```diff
@@ -0,0 +1,119 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+import ProductEventMap from '@pages/product/product-event-map';
+import {EventEmitter} from "events";
+
+const ProductCategoryMap = ProductMap.categories;
+
+export default class Tags {
+  eventEmitter: EventEmitter;
+
+  container: HTMLElement;
+
+  constructor(
+    eventEmitter: EventEmitter,
+    containerSelector: string,
+  ) {
+    this.eventEmitter = eventEmitter;
+    const container = document.querySelector(containerSelector);
+
+    if (!(container instanceof HTMLElement)) {
+      throw '"container" must be a valid HTMLElement';
+    }
+
+    this.container = container;
+  }
+
+  public render(categories: Array<Category>): void {
+    this.container.innerHTML = '';
+
+    const tagTemplate = this.container.dataset.prototype;
+    const prototypeName = this.container.dataset.prototypeName;
+
+    if (!tagTemplate || !prototypeName) {
+      throw 'Tags prototype template or name is undefined or invalid';
+    }
+
+    categories.forEach((category) => {
+      const template = tagTemplate.replace(RegExp(prototypeName, 'g'), String(category.id));
+      const tplFragment = document.createRange().createContextualFragment(template.trim());
+
+      if (tplFragment && tplFragment.firstChild && tplFragment.firstChild.parentNode) {
+        const frag = tplFragment.firstChild.parentNode;
+        const defaultCategoryCheckbox = frag.querySelector(ProductCategoryMap.defaultCategoryCheckbox) as HTMLInputElement;
+        defaultCategoryCheckbox.checked = category.isDefault
+
+        // don't render the tag removal element for main category
+        if (category.isDefault) {
+          const tagRemoveBtn = frag.querySelector(ProductCategoryMap.tagRemoveBtn);
+
+          if (tagRemoveBtn) {
+            tagRemoveBtn.remove();
+          }
+        }
+
+        const namePreviewElement = frag.querySelector(ProductCategoryMap.categoryNamePreview);
+
+        if (namePreviewElement) {
+          namePreviewElement.innerHTML = category.name;
+        }
+
+        this.container.append(frag);
+      }
+    });
+
+    this.listenTagRemoval();
+    this.toggleContainerVisibility();
+    this.eventEmitter.emit(ProductEventMap.categories.categoriesUpdated)
+  }
+
+  private toggleContainerVisibility(): void {
+    this.container.querySelector(ProductCategoryMap.tagsContainer);
+    this.container.classList.toggle(
```

Comment:
Is this really useful? For two reasons:
- there should always be at least one selected category (the default one) so it can never be empty
- even if it could, the container switching depending on the list content might be disturbing